intermediate_source/ddp_tutorial.rst

```diff
@@ -265,8 +265,8 @@ either the application or the model ``forward()`` method.
         setup(rank, world_size)
 
         # setup mp_model and devices for this process
-        dev0 = rank * 2
-        dev1 = rank * 2 + 1
+        dev0 = (rank * 2) % world_size
```

Comment:
Unfortunately, we cannot do this. This will make rank0 and rank 1 share the two GPUs, if there are just two GPUs. DDP and collective comm requires each process to exclusively work on GPUs, otherwise, the comm might hang.

Reply:
The issue I got is like below:

I tried to run the code on 2 GPUs machine.
For rank0 -- dev0=0 dev1=1 
For rank1 --- dev0=2 dev2=3 

With that I got exception about wrong device ordinals since only cuda:0 and cuda:1 are valid.  But for rank2, it ends up with cuda:2 and cuda:3 

Any idea how to avoid such exception?

Reply:
This tutorial will require 8 GPUs to run, as highlighted in the note block in the summary section: "The code in this tutorial runs on an 8-GPU server, but it can be easily generalized to other environments."

Maybe we can error out when the number of GPUs is less than 8?

Reply:
2nd thought, I think only the model parallel part needs 8 GPUs. `demo_model_parallel` we can if just that one when the number of GPUs is less than 8